packages/core/src/routes/profile/index.ts

```diff
@@ -47,22 +48,37 @@ export default function profileRoutes<T extends UserRouter>(
         name: z.string().nullable().optional(),
         avatar: z.string().url().nullable().optional(),
         username: z.string().regex(usernameRegEx).optional(),
+        profile: userProfileGuard.optional(),
       }),
       response: userProfileResponseGuard.partial(),
       status: [200, 400, 422],
     }),
     async (ctx, next) => {
       const { id: userId, scopes } = ctx.auth;
       const { body } = ctx.guard;
-      const { name, avatar, username } = body;
+      const { name, avatar, username, profile } = body;
+      const { address, ...restProfile } = profile ?? {};
 
       assertThat(scopes.has(UserScope.Profile), 'auth.unauthorized');
 
       if (username !== undefined) {
         await checkIdentifierCollision({ username }, userId);
       }
 
-      const updatedUser = await updateUserById(userId, { name, avatar, username });
+      const updatedUser = await updateUserById(userId, {
+        name,
+        avatar,
+        username,
+        ...conditional(
+          profile && {
+            profile: {
+              ...restProfile,
+              // Remove address from profile if user does not have the address scope
+              ...conditional(scopes.has(UserScope.Address) && { address }),
```

Comment:
Should we throw a 403 instead of silently updating other fields?

Reply:
updated